File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_curr = float('inf')
        max_profit = 0
        for i in range(len(prices)):
            min_curr = min(prices[i], min_curr)
            max_profit = max(max_profit, prices[i] - min_curr)
        return max_profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int minCurr = Integer.MAX_VALUE;
        for(int i  = 0 ; i < prices.length;i++){
            minCurr = Math.min(minCurr, prices[i]);
            maxProfit = Math.max(maxProfit,prices[i] -  minCurr );
        }
        return maxProfit;
    }
}